ledger-core/v2/virtual/execute/execute.go

```diff
@@ -416,6 +424,32 @@ func (s *SMExecute) stepSaveNewObject(ctx smachine.ExecutionContext) smachine.St
 		}
 	}
 
+	if s.migrationHappened {
+		return ctx.Jump(s.stepSendDelegatedRequestFinished)
+	}
+
+	return ctx.Jump(s.stepSendCallResult)
+}
+
+func (s *SMExecute) stepSendDelegatedRequestFinished(ctx smachine.ExecutionContext) smachine.StateUpdate {
+	msg := payload.VDelegatedRequestFinished{
+		CallType:           s.Payload.CallType,
+		CallFlags:          s.Payload.CallFlags,
+		Callee:             s.execution.Object,
+		ResultFlags:        nil,
+		CallOutgoing:       s.Payload.CallOutgoing,
+		CallIncoming:       reference.Local{},
+		EntryHeadHash:      nil,
+		DelegationSpec:     nil,
+		DelegatorSignature: nil,
+		ObjectBody:         s.executionNewState.Result.Memory,
```

Comment:
Why do we send it unconditionally? I believe we should send new data only when a new state is produced. I'm sure that object's memory is not enough. At least it should contain identifier of the new state. More should be learned from VStateReport.

Reply:
1. add check if execution is unordered - don't send object body 
2. add ObjectState to VDelegatedRequestFinished